File: backend/app/models/property.rb
class Property < ApplicationRecord
  belongs_to :organization
  belongs_to :branch, optional: true
  belongs_to :owner, optional: true
  has_many :units, dependent: :destroy
  has_many_attached :photos

  enum :property_type, { apartment: 0, hotel: 1, house: 2, hostel: 3 }, validate: true

  normalizes :name, with: ->(v) { v.to_s.strip }
  normalizes :address, with: ->(v) { v.to_s.strip }

  validates :name, presence: true, length: { maximum: 255 }
  validates :address, presence: true, length: { maximum: 500 }
  validates :property_type, presence: true
  validates :description, length: { maximum: 5000 }, allow_blank: true

  validate :branch_must_exist_in_org

  private

  def branch_must_exist_in_org
    return if branch_id.blank?
    return if branch.present? && branch.organization_id == organization_id

    errors.add(:branch, "must exist")
  end
end

File: frontend/src/schemas/property.js
// FT-036 P2: Property schema. Same fields для create + update.
import { z } from 'zod'
import { validate } from './auth'

export const PROPERTY_TYPES = ['apartment', 'hotel', 'house', 'hostel']

export const propertySchema = z.object({
  name: z
    .string({ required_error: 'common.validation.required' })
    .min(1, 'common.validation.required'),
  address: z
    .string({ required_error: 'common.validation.required' })
    .min(1, 'common.validation.required'),
  property_type: z
    .enum(PROPERTY_TYPES, { required_error: 'common.validation.required' }),
  description: z.string().max(5000, 'common.validation.maxLength5000').optional().or(z.literal('')),
  branch_id: z.number().int().nullable().optional(),
})

export { validate }
